slither/tools/read_storage/__main__.py

```diff
@@ -157,6 +221,21 @@ def main() -> None:
         assert args.rpc_url
         srs.walk_slot_info(srs.get_slot_values)
 
+    if args.diff_block:
+        orig_block = srs.block
+        slots_block_orig = copy.deepcopy(srs.slot_info)
+
+        srs.block = diff_block
+        # This takes into account differences in dynamic arrays
+        if args.variable_name:
+            srs.get_target_variables(**vars(args))
+        else:
+            srs.get_storage_layout()
+        srs.walk_slot_info(srs.get_slot_values)
+        slots_block_diff = copy.deepcopy(srs.slot_info)
+
+        print_diff_table(orig_block, diff_block, slots_block_orig, slots_block_diff)
```

Comment:
Can you move this logic out of __main__ and do something like `srs.walk_slot_info(srs.convert_slot_info_to_rows)`